[webkitapipy] OBJC_CLASS symbols not being tracked as classes

emw-apple opened the pull request:
#### ce95a45d719f9337a4219ea98dea360671780b68
<pre>
[webkitapipy] OBJC_CLASS symbols not being tracked as classes
<a href="https://bugs.webkit.org/show_bug.cgi?id=294639">https://bugs.webkit.org/show_bug.cgi?id=294639</a>

Reviewed by Alexey Proskuryakov.

Small bugfix which causes audit-api to recognize ObjC classes in
binaries it imports, such that using those classes in other libraries
does not automatically count as SPI use.

As a drive-by, fix up type hints in sdkdb.py to be accurate and conform
to webkitapipy&apos;s line widths.

* Tools/Scripts/libraries/webkitapipy/webkitapipy/sdkdb.py:
(SDKDB.add_binary):
(SDKDB.add_tbd):
(SDKDB._add_objc_interface):
(SDKDB._add_objc_selector):

Canonical link: <a href="https://commits.webkit.org/296404@main">https://commits.webkit.org/296404@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0818d9127991e0cd00ce777526f045ff1cdf55b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108264 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27925 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18347 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113473 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58713 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110227 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28619 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36476 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82200 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111212 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22679 "Found 1 new test failure: http/tests/geolocation/geolocation-get-current-position-does-not-leak.https.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97517 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62633 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/107696 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22092 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15652 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58206 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92045 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15708 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116595 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35323 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26007 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91227 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35695 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93791 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91024 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23226 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35918 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13676 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31084 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35223 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34945 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38299 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36621 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0818d9127991e0cd00ce777526f045ff1cdf55b1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46874

aproskuryakov reviewed (approved):
This makes me sad about not having regression tests for this code.

emw-apple committed:
[webkitapipy] OBJC_CLASS symbols not being tracked as classes
https://bugs.webkit.org/show_bug.cgi?id=294639

Reviewed by Alexey Proskuryakov.

Small bugfix which causes audit-api to recognize ObjC classes in
binaries it imports, such that using those classes in other libraries
does not automatically count as SPI use.

As a drive-by, fix up type hints in sdkdb.py to be accurate and conform
to webkitapipy's line widths.

* Tools/Scripts/libraries/webkitapipy/webkitapipy/sdkdb.py:
(SDKDB.add_binary):
(SDKDB.add_tbd):
(SDKDB._add_objc_interface):
(SDKDB._add_objc_selector):

Canonical link: https://commits.webkit.org/296404@main

emw-apple commented:
> This makes me sad about not having regression tests for this code.

Thank you for the inspiration! https://github.com/WebKit/WebKit/pull/46939
